lib/rfxcom.js

```diff
@@ -66,50 +103,94 @@ util.inherits(RfxCom, EventEmitter);
 RfxCom.prototype.open = function() {
     var self = this;
 
-    if (typeof self.serialport === "undefined") {
-        self.serialport = new serialport.SerialPort(self.device, {
-            baudrate: 38400,
-            parser: self.rfxtrxParser()
+    if (typeof self.serialport === "undefined" || self.fakingSerialPort) {
```

Comment:
The basic mechanics - which is just how you designed it - is that in normal use, the RfxCom object creates a serialport based on the path in the first constructor argument, but in testing use the port property of the second argument is a ready made FakeSerialPort. This probably means that you are right, and the condition at new line 106 is always true, though I haven't yet tested this. I'll look into it, although that last 'Bump!" comment unleashed upon me a whole deluge of 5-minute jobs which had been lurking waiting for me to think I had some spare time...

If this is the case, then if the FakeSerialPort had an open() method (which doesn't need to do anything) we could lose the fakingSerialPort property entirely, which would be nice.

I found that, if the USB cable gets disconnected, after emitting the 'error' or 'disconnect' event, the serialport object is unable to recover or be re-opened when the cable gets reconnected. The disconnectHandler calls RfxCom.close(), which closes the serialport (if needs closing, i.e. it still has a non-falsy file descriptor) and then destroys it. A fresh new serialport is created on the next call to RfxCom.open().

This means that you can't easily pass in a serialport object to the RfxCom constructor, at least not if you treat it as an opaque object. You could retrieve its path name, and use that to recreate the serialport if necessary after a disconnect, but in that case it seems simpler just to pass in the path name as now, and let RfxCom handle the actual Serialport object creation & destruction.